### ISSUE TITLE
incanter.io/read-dataset converts negative integers to floats when reading csv files

### ISSUE THREAD

**Issue by Engelberg** [incanter/incanter#331]
To reproduce, create a csv file with a mixture of positive and negative integers.  The positive integers get read in as integers, but the negative integers turn into floats.

**Comment by Engelberg** [comment]
Found the problem.  In incanter.io, the definition of parse-string is:

```clojure
(defn- parse-string [value & [empty-value]]
  (if (= value "")
    empty-value
    (if (re-matches #"\d+" value)
      (try (Long/parseLong value)
           (catch NumberFormatException _ value))
      (try (Double/parseDouble value)
           (catch NumberFormatException _ value)))))
```

Negative integers don't match this regex.  This is not the correct regex for integers.
Probably something like
#"-?\d+"

**Comment by pho-coder** [comment]
If I want to get ingeter string, this funcation also don't work.
I can't get "00001", only got 1.

**Comment by alexott** [comment]
fixed by commit 4d6af4a
